source/grid/grid_generator.cc

```diff
@@ -5845,6 +5845,233 @@ namespace GridGenerator
+              // 0 1   0 3
+              //
+              // by forcing the second vertex (in lexical ordering) to be
+              // smaller than the third
+              if (renumbering[1] > renumbering[2])
+                std::swap(renumbering[1], renumbering[2]);
+              std::copy(renumbering.begin(),
+                        renumbering.end(),
+                        std::begin(cell_data.vertices));
+            }
+        }
+
+      // Now that all cell objects have been canonicalized they can be sorted:
+      auto compare = [](const CellData<structdim> &a,
+                        const CellData<structdim> &b) {
+        return std::lexicographical_compare(std::begin(a.vertices),
+                                            std::end(a.vertices),
+                                            std::begin(b.vertices),
+                                            std::end(b.vertices));
+      };
+      std::sort(subcell_data.begin(), subcell_data.end(), compare);
+
+      // Finally, determine which objects are duplicates. Duplicates are
+      // assumed to be interior objects, so delete all but one and change the
+      // boundary id:
+      auto left = subcell_data.begin();
+      while (left != subcell_data.end())
+        {
+          const auto right =
+            std::upper_bound(left, subcell_data.end(), *left, compare);
+          // if the range has more than one item then there are duplicates -
+          // set all boundary ids in the range to the internal boundary id
+          if (left + 1 != right)
+            for (auto it = left; it != right; ++it)
+              {
+                it->boundary_id = numbers::internal_face_boundary_id;
+                Assert(it->manifold_id == left->manifold_id,
+                       ExcMessage(
+                         "In the process of grid generation a single "
+                         "line or quadrilateral has been assigned two "
+                         "different manifold ids. This can happen when "
+                         "a Triangulation is copied, e.g., via "
+                         "GridGenerator::replicate_triangulation() and "
+                         "not all external boundary faces have the same "
+                         "manifold id. Double check that all faces "
+                         "which you expect to be merged together have "
+                         "the same manifold id."));
+              }
+          left = right;
+        }
+
+      const auto duplicates_begin =
+        std::unique(subcell_data.begin(), subcell_data.end());
+      subcell_data.resize(duplicates_begin - subcell_data.begin());
+    }
+  } // namespace
+
+
+
+  template <int dim, int spacedim>
+  void
+  replicate_triangulation(const Triangulation<dim, spacedim> &input,
+                          const std::vector<unsigned int> &   extents,
+                          Triangulation<dim, spacedim> &      result)
+  {
+    AssertDimension(dim, extents.size());
```

Comment:
Why don't we use `std::array` if the size is fixed?

Reply:
I would prefer that but we consistently use `std::vector<unsigned int>` for stuff like this in GridGenerator - I am just using the same convention.